thrust/thrust/iterator/detail/iterator_category_to_system.h

```diff
@@ -36,27 +35,18 @@ THRUST_NAMESPACE_BEGIN
 
 namespace detail
 {
-template <typename T>
-_CCCL_INLINE_VAR constexpr bool is_iterator_system =
-  ::cuda::std::is_convertible_v<T, any_system_tag> || ::cuda::std::is_convertible_v<T, host_system_tag>
-  || ::cuda::std::is_convertible_v<T, device_system_tag>;
+auto category_to_system_impl(...) -> void;
+
+auto category_to_system_impl(const input_host_iterator_tag&) -> host_system_tag;
+auto category_to_system_impl(const output_host_iterator_tag&) -> host_system_tag;
+
+auto category_to_system_impl(const input_device_iterator_tag&) -> device_system_tag;
+auto category_to_system_impl(const output_device_iterator_tag&) -> device_system_tag;
```

Comment:
Those are all missing `_CCCL_HOST_DEVICE`

Reply:
Even though they are just declarations used only in constant evaluation?

Reply:
It's NVRTC ...